src/main/java/com/esl/demo/controller/ESLController.java

```diff
@@ -1,35 +1,94 @@
 package com.esl.demo.controller;
 
+import com.esl.demo.dto.ESLDto;
+import com.esl.demo.rest.errors.BadRequestException;
+import com.esl.demo.rest.errors.ErrorConstants;
+import com.esl.demo.rest.errors.ResourceNotFoundException;
+import com.esl.demo.service.interfaces.ESLService;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
 @RequestMapping("/api/esls")
-public class ESLController implements AbstractController{
+public class ESLController implements AbstractController<ESLDto, Long> {
 
+    private final ESLService eslService;
+
+    public ESLController(ESLService eslService) {
+        this.eslService = eslService;
+    }
+
+    @GetMapping("/{id}")
```

Comment:
Add java doc - comments

Reply:
Endpoints remain restfull without any special case. As such it can be expected to be considered "Self documenting" :)